src/desktop/input_capture.rs

```diff
@@ -1,3 +1,269 @@
+//!         )
+//!         .await?;
+//!
+//!     // connect to eis server
+//!     eprintln!("connect_to_eis");
+//!     let fd = input_capture.connect_to_eis(&session).await?;
+//!
+//!     // create unix stream from fd
+//!     let stream = UnixStream::from(fd);
+//!     stream.set_nonblocking(true)?;
+//!
+//!     // create ei context
+//!     let context = ei::Context::new(stream)?;
+//!     context.flush().unwrap();
+//!
+//!     let mut event_stream = EiEventStream::new(context.clone())?;
+//!     let interfaces = INTERFACES.get_or_init(|| {
+//!         let mut m = HashMap::new();
+//!         m.insert("ei_connection", 1);
+//!         m.insert("ei_callback", 1);
+//!         m.insert("ei_pingpong", 1);
+//!         m.insert("ei_seat", 1);
+//!         m.insert("ei_device", 2);
+//!         m.insert("ei_pointer", 1);
+//!         m.insert("ei_pointer_absolute", 1);
+//!         m.insert("ei_scroll", 1);
+//!         m.insert("ei_button", 1);
+//!         m.insert("ei_keyboard", 1);
+//!         m.insert("ei_touchscreen", 1);
+//!         m
+//!     });
+//!     let response = reis::tokio::ei_handshake(
+//!         &mut event_stream,
+//!         "ashpd-mre",
+//!         ei::handshake::ContextType::Receiver,
+//!         interfaces,
+//!     )
+//!     .await
+//!     .expect("ei handshake failed");
+//!
+//!     let mut event_stream = EiConvertEventStream::new(event_stream, response.serial);
+//!
+//!     let pos = Position::Left;
+//!     let zones = input_capture.zones(&session).await?.response()?;
+//!     eprintln!("zones: {zones:?}");
+//!     let barriers = zones
+//!         .regions()
+//!         .iter()
+//!         .enumerate()
+//!         .map(|(n, r)| {
+//!             let id = n as u32;
+//!             let (x, y) = (r.x_offset(), r.y_offset());
+//!             let (width, height) = (r.width() as i32, r.height() as i32);
+//!             let barrier_pos = match pos {
+//!                 Position::Left => (x, y, x, y + height - 1), // start pos, end pos, inclusive
+//!                 Position::Right => (x + width, y, x + width, y + height - 1),
+//!                 Position::Top => (x, y, x + width - 1, y),
+//!                 Position::Bottom => (x, y + height, x + width - 1, y + height),
+//!             };
+//!             Barrier::new(id, barrier_pos)
+//!         })
+//!         .collect::<Vec<_>>();
+//!
+//!     eprintln!("requested barriers: {barriers:?}");
+//!
+//!     let request = input_capture
+//!         .set_pointer_barriers(&session, &barriers, zones.zone_set())
+//!         .await?;
+//!     let response = request.response()?;
+//!     let failed_barrier_ids = response.failed_barriers();
+//!
+//!     eprintln!("failed barrier ids: {:?}", failed_barrier_ids);
+//!
+//!     input_capture.enable(&session).await?;
+//!
+//!     let mut activate_stream = input_capture.receive_activated().await?;
+//!
+//!     loop {
+//!         let activated = activate_stream.next().await.unwrap();
```

Comment:
```suggestion
//!         let activated = activate_stream.next().await?;
```

Reply:
this does not work, stream returns an option and I could otherwise not return ashpd::Result

Reply:
Ah indeed, my bad.